providers/gitlab/provider.go

```diff
@@ -0,0 +1,73 @@
+package gitlab
+
+import (
+	"context"
+	"strings"
+
+	"github.com/stamm/dep_radar"
+	"github.com/xanzy/go-gitlab"
+)
+
+var (
+	_ dep_radar.ITagGetter = &Provider{}
+	_ dep_radar.IProvider  = &Provider{}
+)
+
+// Provider for gitlab
+type Provider struct {
+	goGetURL string
+
+	client *gitlab.Client
+}
+
+// New creates new instance of provider
+func New(client *gitlab.Client, goGetURL string) *Provider {
+	return &Provider{
+		client:   client,
+		goGetURL: goGetURL,
+	}
+}
+
+func (p *Provider) cleanID(pkg dep_radar.Pkg) string {
+	pack := strings.TrimPrefix(string(pkg), p.goGetURL)
+	pack = strings.Trim(pack, "/")
+	return pack
+}
+
+// File gets file from github
+func (p *Provider) File(ctx context.Context, pkg dep_radar.Pkg, branch, filename string) ([]byte, error) {
+	opts := &gitlab.GetRawFileOptions{
+		Ref: &branch,
+	}
+	data, _, err := p.client.RepositoryFiles.GetRawFile(p.cleanID(pkg), filename, opts)
+	if err != nil {
+		return nil, err
+	}
+
+	return data, nil
+}
+
+// GoGetURL gets url for go get
+func (p *Provider) GoGetURL() string {
+	return p.goGetURL
+}
+
+// Tags get tags from github
+func (p *Provider) Tags(ctx context.Context, pkg dep_radar.Pkg) ([]dep_radar.Tag, error) {
+	opts := gitlab.ListTagsOptions{}
+	tagsResp, _, err := p.client.Tags.ListTags(p.cleanID(pkg), &opts)
```

Comment:
I think we need to set PerPage options go get more tags. I have bug in my implementation for other hostings, that I don't walk through all pages

Reply:
Having studied the code, I still do not understand why you are downloading all versions of the library?

Reply:
Because you may need it for another application dependencies